### ISSUE TITLE
New version not created when publish is true and runtime is modified

### ISSUE THREAD

**Issue by bassmanitram** [terraform-providers/terraform-provider-aws#11592]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
 0.11.14

### Affected Resource(s)

* aws_lambda_function

### Terraform Configuration Files

```hcl
resource "aws_lambda_function" "lambda_vrqst" {
  
  provider = "aws.us-east-1"

  filename         = "${data.external.artifact_data.result.output_file}"
  source_code_hash = "${base64sha256(file(data.external.artifact_data.result.output_file))}"

  function_name    = "${var.environment_name}-${var.aws_region}-ui-vrequest-${random_string.function_name_suffix.result}"
  role             = "${var.lambda_exec_role_arn}"
  handler          = "${var.vrqst_lambda_handler}"
  runtime          = "${var.lambda_runtime}"
  memory_size      = "${var.vrqst_lambda_memory_size}"
  timeout          = "${var.vrqst_lambda_timeout}"

  publish          = "true"

  tags = "${merge(map("Name", "${var.environment_name}-${var.aws_region}-ui-vrequest"), var.resource_tags)}"
}
```

### Expected Behavior

var.lambda_runtime was changed from nodejs12.x to nodejs10.x (Edge Lambdas don't work with 12.x) and the plan recreated and applied.

Expected version 2 of the lambda to be published

### Actual Behavior

No new version published

* #0000

**Comment by jbergknoff-rival** [comment]
Related: https://github.com/terraform-providers/terraform-provider-aws/pull/11211

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.50.0 of the Terraform AWS Provider, later today. Thanks to @nemreid for the implementation. 👍 

Please note that this resource will now use a separate `PublishVersion` API call, so any restrictive IAM policies for the Terraform run may require updates.
